src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1355,6 +1377,111 @@ def _deny_request(
     )
 
 
+@router.post(
+    PRIVACY_REQUEST_PRE_APPROVE_ELIGIBLE,
+    status_code=HTTP_200_OK,
+)
+def mark_privacy_request_pre_approve_eligible(
+    privacy_request_id: str,
+    *,
+    db: Session = Depends(deps.get_db),
+    config_proxy: ConfigProxy = Depends(deps.get_config_proxy),
+    webhook: PreApprovalWebhook = Security(
+        verify_callback_oauth_pre_approval_webhook, scopes=[PRIVACY_REQUEST_REVIEW]
+    ),
+) -> None:
+    """
+    Marks privacy request as eligible for automatic approval.
+    If all webhook responses have been received and all are affirmative, proceed to queue privacy request
+    """
+    PreApprovalWebhookReply.create(
+        db=db,
+        data={
+            "webhook_id": webhook.id,
+            "privacy_request_id": privacy_request_id,
+            "is_eligible": True,
+        },
+    )
+
+    all_webhooks = PreApprovalWebhook.all(db)
+    if not all_webhooks:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"No configured webhooks found. Could not pre-approve privacy request '{privacy_request_id}'.",
+        )
+    replies_for_privacy_request = PreApprovalWebhookReply.filter(
+        db=db,
+        conditions=(PreApprovalWebhookReply.privacy_request_id == privacy_request_id),
+    ).all()
+
+    # Check if not all replies have been received
+    if len(replies_for_privacy_request) < len(all_webhooks):
```

Comment:
Since hitting this endpoint multiple times for the same webhook could create multiple replies this could be accidentally satisfied even if we hadn't received responses from all webhooks. I would check verify that each webhook has a corresponding reply instead of relying on a count.